packages/syft/src/syft/core/node/common/client_manager/oblv_api.py

```diff
@@ -0,0 +1,155 @@
+                raise Exception(f"{type(self)} has no response")
+            return content
+
+    def check_connection(self, deployment: DeploymentClient, **kwargs: Any) -> Any:
+        content = {
+            "deployment_id": deployment.deployment_id,
+            "client": SyftOblvClient.from_client(deployment.oblv_client),
+        }
+        response = self.perform_api_request(
+            syft_msg=CheckEnclaveConnectionMessage, content=content
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            content = getattr(response, "resp_msg")
+            if content is None:
+                raise Exception(f"{type(self)} has no response")
+            return content
+
+    def send_dataset(
+        self,
+        deployment: DeploymentClient,
+        dataset: Union[str, TensorWrappedPhiTensorPointer],
+        **kwargs: Any,
+    ) -> Any:
+        content = {
+            "deployment_id": deployment.deployment_id,
+            "client": SyftOblvClient.from_client(deployment.oblv_client),
+        }
+        dataset_id = (
+            dataset if type(dataset) == "str" else dataset.id_at_location.to_string()
+        )
+        content.update({"dataset_id": dataset_id})
+        response = self.perform_api_request(
+            syft_msg=PublishDatasetMessage, content=content
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            dataset_id = getattr(response, "dataset_id")
+            return OblvTensorWrapper(id=dataset_id, deployment_client=deployment)
+
+    def publish_budget(
+        self, deployment_id, publish_request_id, client, **kwargs: Any
+    ) -> Any:
+        response = self.oblv_perform_api_request_without_reply(
+            syft_msg=PublishApprovalMessage,
+            content={
+                "result_id": publish_request_id,
+                "client": SyftOblvClient.from_client(client),
+                "deployment_id": deployment_id,
+            },
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+
+    def publish_request_budget_deduction(
+        self, deployment_id, publish_request_id, budget_to_deduct, client, **kwargs: Any
+    ) -> Any:
+        response = self.oblv_perform_api_request_without_reply(
+            syft_msg=DeductBudgetMessage,
+            content={
+                "result_id": publish_request_id,
+                "client": SyftOblvClient.from_client(client),
+                "budget_to_deduct": budget_to_deduct,
+                "deployment_id": deployment_id,
+            },
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+
+    def __getitem__(self) -> Any:
+        return self.get()
+
+    def oblv_perform_api_request_without_reply(
+        self,
+        syft_msg: Optional[Type[SyftMessage]],
```

Comment:
Q: Is this optional in the sense that we might send none or we should initialize it with None?

Reply:
Solved by [930bba3](https://github.com/OpenMined/PySyft/pull/7202/commits/930bba34747b97eec3fef12c388e12db60c7e7f4)